packages/plugin-pnpm/sources/PnpmLinker.ts

```diff
@@ -92,10 +103,12 @@ class PnpmInstaller implements Installer {
   }
 
   async installPackageHard(pkg: Package, fetchResult: FetchResult, api: InstallPackageExtraApi) {
-    const pkgPath = getPackageLocation(pkg, {project: this.opts.project});
+    // We only enable self-references if the package doesn't depend on a different
+    // version of itself (in which case it should resolve to that version)
+    const pkgPath = getPackageLocation(pkg, {project: this.opts.project, createSelfReference: !pkg.dependencies.has(pkg.identHash)});
 
     this.customData.locatorByPath.set(pkgPath, structUtils.stringifyLocator(pkg));
-    this.packageLocations.set(pkg.locatorHash, pkgPath);
+    this.customData.packageLocations.set(pkg.locatorHash, pkgPath);
```

Comment:
What do you think about calling it `pathByLocator`? I didn't realize it when I first wrote this, but given that those two maps will now be in the same store with exactly mirrored semantics, I feel like they should have more similar names.

Reply:
Done :+1: